### ISSUE TITLE
Three enhancement proposals

### ISSUE THREAD

**Issue by bouscram** [teras/Jubler#12]
I have been using Jubler a lot for many years (many thanks for this program), and there are three small things I'd like to have:

- The ability to define a default encoding for saved files (I'd like it to be UTF-8 to get rid of undisplayable characters like "œ" in ISO Latin-1)
- The possibility to set the default setting for reencoding, time shifting, ... : for example, 90% of the time I do reencoding from 23,976 FPS to 25 FPS globally (to adapt dowloaded US subtitles to French broadcast capture), so I would like to have the selection set to time interval, from 0:00:... to 24:00:.. by default
- The possibility to redo the last edit. For example, it would be very useful when shifting time: you add 200 ms to fine tune the shift, then again, etc. until it's good

**Comment by bouscram** [comment]
Hello - I had forgotten this one...

Just a few explanations:

for 2 : I'm not talking about frames, but the possibility to set the default setting of all the timing modification tools to the "by time" setting (and set the min and max time values), instead of the "user selection". Most of the time, my changes affect all the subtitles, so setting the min time value to 0:00 and the max value to 24:00 is the most convenient.

for 3 : I'm talking about a **redo**, not **undo**, to repeat the last change.

I hope it's clearer now. Anyway, I can perfectly use Jubler as it is, it's really a very useful tool ;-)

**Comment by teras** [comment]
2) Oh now I get it. Indeed that should be the default I agree. Me too, most of the tasks usually affect the whole subtitle range. 
I have an idea: if the selected subtitles are more than one, then the "selected" to be the default, if only one then "all" to be the default. What do you think?

3) "Undo" doesn't remember the action, it remembers the state. It is independent of *how* we get there. For this reason "Redo" might not be straight forward how to implement it. I can't think of a systematic way, except going to each tool by itself and think of a way that makes sense. 
For example,  going from top to bottom, 

* split file, what where?
* shift time (maybe only this makes sense)
* recode... again and again the same amount? I can't think of a scenario that can be used more than once
* time fix, you fixed it already
* round time, also once is enouigh
* spell check... how many times should you spell check?
* translate.. can't translate something that is already translated
* join/split,just do it again

For all other tools, you can simply assign a shortcut and execute the shortcut again. Actually all menu entries can have shortcuts

**Comment by teras** [comment]
About (2).

When looking back the code I saw that there was an easier solution to this anyway. The dialog always remembers the last position of the selection, and when the user shows another dialog, this selection is applied to the new selection too.

What didn't happen, and maybe that's why this feature wasn't obvious, is that this behavior wasn't saved across Jubler sessions. So I believe the best solution to this is simply to save the section across sessions.

**Comment by teras** [comment]
If you agree, can we close this issue?

**Comment by bouscram** [comment]
Yes sure ! Do you have an idea of when you will release a new version?
